From debf069a302f99b35efe11d08b01b5da39b88965 Mon Sep 17 00:00:00 2001
From: Dan Williams <dan.j.williams@intel.com>
Date: Fri, 10 Jun 2016 18:20:53 -0700
Subject: nfit: don't override return value of nfit_mem_init
References: FATE#321390
Git-commit: 81ed4e3670853e4cebad88aeffc0ba1d90d4d6ed
Patch-mainline: v4.8-rc1

We were needlessly converting nfit_mem_init() errors to -ENOMEM.

Signed-off-by: Dan Williams <dan.j.williams@intel.com>
Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>

---
 drivers/acpi/nfit.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/acpi/nfit.c b/drivers/acpi/nfit.c
index 48fdc1b..ce3078a 100644
--- a/drivers/acpi/nfit.c
+++ b/drivers/acpi/nfit.c
@@ -2533,10 +2533,9 @@ int acpi_nfit_init(struct acpi_nfit_desc *acpi_desc, void *data, acpi_size sz)
 	if (rc)
 		goto out_unlock;
 
-	if (nfit_mem_init(acpi_desc) != 0) {
-		rc = -ENOMEM;
+	rc = nfit_mem_init(acpi_desc);
+	if (rc)
 		goto out_unlock;
-	}
 
 	rc = acpi_nfit_register_dimms(acpi_desc);
 	if (rc)
-- 
1.8.5.6

